sklearn/tests/test_calibration.py

```diff
@@ -42,61 +42,71 @@ def test_calibration():
     clf = MultinomialNB().fit(X_train, y_train, sample_weight=sw_train)
     prob_pos_clf = clf.predict_proba(X_test)[:, 1]
 
-    pc_clf = CalibratedClassifierCV(clf, cv=y.size + 1)
-    assert_raises(ValueError, pc_clf.fit, X, y)
-
-    # Naive Bayes with calibration
-    for this_X_train, this_X_test in [(X_train, X_test),
-                                      (sparse.csr_matrix(X_train),
-                                       sparse.csr_matrix(X_test))]:
-        for method in ['isotonic', 'sigmoid']:
-            pc_clf = CalibratedClassifierCV(clf, method=method, cv=2)
-            # Note that this fit overwrites the fit on the entire training
-            # set
-            pc_clf.fit(this_X_train, y_train, sample_weight=sw_train)
-            prob_pos_pc_clf = pc_clf.predict_proba(this_X_test)[:, 1]
-
-            # Check that brier score has improved after calibration
-            assert (brier_score_loss(y_test, prob_pos_clf) >
-                           brier_score_loss(y_test, prob_pos_pc_clf))
-
-            # Check invariance against relabeling [0, 1] -> [1, 2]
-            pc_clf.fit(this_X_train, y_train + 1, sample_weight=sw_train)
-            prob_pos_pc_clf_relabeled = pc_clf.predict_proba(this_X_test)[:, 1]
-            assert_array_almost_equal(prob_pos_pc_clf,
-                                      prob_pos_pc_clf_relabeled)
-
-            # Check invariance against relabeling [0, 1] -> [-1, 1]
-            pc_clf.fit(this_X_train, 2 * y_train - 1, sample_weight=sw_train)
-            prob_pos_pc_clf_relabeled = pc_clf.predict_proba(this_X_test)[:, 1]
-            assert_array_almost_equal(prob_pos_pc_clf,
-                                      prob_pos_pc_clf_relabeled)
-
-            # Check invariance against relabeling [0, 1] -> [1, 0]
-            pc_clf.fit(this_X_train, (y_train + 1) % 2,
-                       sample_weight=sw_train)
-            prob_pos_pc_clf_relabeled = \
-                pc_clf.predict_proba(this_X_test)[:, 1]
-            if method == "sigmoid":
-                assert_array_almost_equal(prob_pos_pc_clf,
-                                          1 - prob_pos_pc_clf_relabeled)
-            else:
-                # Isotonic calibration is not invariant against relabeling
-                # but should improve in both cases
+    # Testing both ensemble approach and single estimator approach
+    for ensemble in (True, False):
```

Comment:
I haven't checked the test but we might want to use parametrization instead (see below).